SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1151,38 +1151,34 @@ - (void)handleSendToMyself:(TSOutgoingMessage *)outgoingMessage
     // create a new transaction to avoid deadlock.
     NSString *contactId = [TSAccountManager localNumber];
     [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
-        TSContactThread *cThread =
-            [TSContactThread getOrCreateThreadWithContactId:contactId transaction:transaction];
+        TSContactThread *cThread = [TSContactThread getOrCreateThreadWithContactId:contactId transaction:transaction];
         [cThread saveWithTransaction:transaction];
 
         // We need to clone any attachments for message sent to self; otherwise deleting
         // the incoming or outgoing copy of the message will break the other.
-        NSMutableArray<NSString *> *attachmentIds = [NSMutableArray new];
+        NSMutableArray<NSString *> *incomingAttachmentIds = [NSMutableArray new];
```

Comment:
We were already cloning "body" attachments; we now use `cloneAttachmentForSendToMyself` to do this.